book/text.md

```diff
@@ -748,6 +748,75 @@ styles. A particular challenge is [Mongolian script][mongolian].
 [vertical]: https://www.smashingmagazine.com/2019/08/writing-modes-layout/
 [mongolian]: https://www.w3.org/TR/mlreq/
 
+Faster text layout
+==================
+
+Now that you've implemented styled text, you've probably
+noticed---unless you're on macOS[^macos-cache]---that on a large web
+page like this chapter your browser has slowed significantly from the
+[last chapter](graphics.md). That's because text layout, and
+specifically the part where you measure each word, is quite slow.
+
+[^macos-cache]: The macOS text APIs do fairly complex system-wide font
+    caching that Linux and Windows do not do. The optimization
+    described in this section won't hurt any.
+
+Unfortunately, it's hard to make text measurement much faster. With
+proportional fonts and complex font features like hinting and kerning,
+measuring text can require pretty complex computations. But on a large
+web page, some words likely appear a lot---for example, this page
+includes the word "the" over two hundred times. Instead of measuring
+these words over and over again, we could measure them once, and then
+cache the results. On normal English text, this usually results in a
+substantial speedup.
+
+Caching is such a good idea that Tk already implements it internally.
+But there's a small catch: Tk stores the cache inside each `Font`
+object. But our `text` method creates a new `Font` object for each
```

Comment:
Did you verify this?

Reply:
Good point. Working on it.

Reply:
Ok—here's what I learned. Creating a `tkinter.font.Font` object the way we do calls the `font create` command in Tk. That calls the `FONT_CREATE` branch in `Tk_FontObjCmd` method in `generic/tkFont.c`. The important thing is that in this process, `tkinter.font.Font` chooses a new name for the font, which causes `Tk_FontObjCmd` to create a new `Tk_Font` object, because the cache is keyed by name. Ok, so now we need to know what the created `Tk_Font` does. This is platform-specific, so we go to `unix/tkUnixFont.c` as an example, and if I follow the code correctly, the underlying X font is created in `GetScreenFont`, with no cache.

So I think what is ultimately happening here is that caching the `tkinter.font.Font` objects reuses the underlying system fonts—`XFontStruct`s and windows `HFONT`s, which I assume ultimately means that system-level caching kicks in. I'll revise the text to explain this better.